bracket-push.json

```diff
@@ -0,0 +1,49 @@
+{
+  "cases": [
+    {
+      "description": "empty string",
+      "input": "",
+      "expected": true
+    },
+    {
+      "description": "unpaired brackets",
+      "input": "[[",
+      "expected": false
+    },
+    {
+      "description": "wrong ordered brackets",
+      "input": "}{",
+      "expected": false
+    },
+    {
+      "description": "paired and nested brackets",
+      "input": "([{}({}[])])",
+      "expected": true
+    },
+    {
+      "description": "unpaired and nested brackets",
+      "input": "([{])",
+      "expected": false
+    },
+    {
+      "description": "paired and wrong nested brackets",
+      "input": "[({]})",
+      "expected": false
+    },
+    {
+      "description": "paired with whitespace",
+      "input": "{ }",
+      "expected": true
+    },
+    {
+      "description": "math expression",
+      "input": "(((185 + 223.85) * 15) - 543)/2",
+      "expected": true
+    },
+    {
+      "description": "complex latex expression",
+      "input": "\\left(\begin{array}{cc} \frac{1}{3} & x\\ \\mathrm{e}^{x} &... x^2 \\end{array}\right)",
```

Comment:
Every `\` that you want to appear in the string as a character must be escaped (i.e., `\\`).
For example, `\b` is backspace, `\f` is form feed, and `\r` is carriage return here.
Getting closer, though :smile: